bin/annotate_test_failures

```diff
@@ -118,8 +118,8 @@ end
 # Given a list of failures for a given step and state, update the corresponding annotation
 #
 def update_annotation(title, list, style, state)
-  step_id = ENV.fetch('BUILDKITE_STEP_ID', title)
-  annotation_context = "step-#{step_id}-#{state}".downcase.gsub(/\W/, '-')
+  ctx_components = ['test-summary', ENV.fetch('BUILDKITE_STEP_ID', title), ENV.fetch('BUILDKITE_PARALLEL_JOB', nil), state]
+  annotation_context = ctx_components.compact.join('-').downcase.gsub(/\W/, '-')
```

Comment:
The `.compact` call ensures to remove any `nil` entry from the `ctx_components` array. That's what makes this include the `BUILDKITE_PARALLEL_JOB` value if it's defined, but skip it if it's `nil`.

<img width="997" alt="image" src="https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/assets/216089/c65b06cd-8c0a-4d17-87da-dc9f3e07417f">